jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cwc/editor/context/file/util/FileContextExtractorTest.kt

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import io.mockk.clearMocks
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import junit.framework.TestCase.assertNull
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import software.aws.toolkits.jetbrains.services.amazonq.webview.FqnWebviewAdapter
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.file.FileContextExtractor
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.file.util.LanguageExtractor
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.file.util.MatchPolicyExtractor
+import software.aws.toolkits.jetbrains.services.cwc.utility.EdtUtility
+
+class FileContextExtractorTest {
+
+    // Constructor parameters
+    private val mockFqnWebviewAdapter: FqnWebviewAdapter = mockk<FqnWebviewAdapter>(relaxed = true)
+    private val mockProject: Project = mockk<Project>(relaxed = true)
+    private val mockLanguageExtractor: LanguageExtractor = mockk<LanguageExtractor>(relaxed = true)
+
+    private val mockEditor: Editor = mockk<Editor>(relaxed = true)
+
+    /*
+    private lateinit var fileEditorManager: FileEditorManager
+    private lateinit var psiDocumentManager: PsiDocumentManager
+     */
+
+    private lateinit var fileContextExtractor: FileContextExtractor
+
+    @Before
+    fun setUp() {
+        fileContextExtractor = FileContextExtractor(mockFqnWebviewAdapter, mockProject, mockLanguageExtractor)
+
+        // Editor
+        mockkStatic(FileEditorManager::class)
```

Comment:
You need to match every call to `mockkStatic`, `mockkObject`, and `mockkConstructor` with a matching call to `unmockkStatic`/`unmockkObject`/`unmockkConstructor`, or simply do an `unmockkAll`. This is because these are static mocks, not just mock instances, and can leak across tests.

This is why I prefer using the `mockkStatic(class) { .... }` form of these functions where possible, which automatically unmocks at the end of the lambda. Since you're doing it in the `@Before` here though you need to unmock in the `@After`

Reply:
Wouldn't I just need to un mock static?  A mockObject would never leak between tests right - unless it's explicitly stored in an instance variable?

I called unmockkAll